lib/rucio/client/uploadclient.py

```diff
@@ -218,6 +220,8 @@ def upload(self, items, summary_file_path=None):
                     if not self.client.update_replicas_states(rse, files=[replica_for_api]):
                         logger.warning('Failed to update replica state')
             else:
+                self.trace['clientState'] = 'FAILED'
```

Comment:
@tbeerman said you need to fill the `self.trace['stateReason']` field. I think you can write the exception from `except (ServiceUnavailable, ResourceTemporaryUnavailable) as error:` into this as a string.